src/js/view/controls/components/tooltipicon.ts

```diff
@@ -3,16 +3,27 @@ import ariaLabel from 'utils/aria';
 import { toggleClass } from 'utils/dom';
 import svgParse from 'utils/svgParser';
 
-export default class TooltipIcon {
-    constructor(name, ariaText, elementShown, svgIcons) {
+interface TooltipIcon extends Events {
+    el: HTMLElement;
+    tooltip: HTMLElement;
+    openClass: string;
+    componentType: string;
+    isOpen?: boolean;
+    content?: HTMLElement | null;
+}
+
+class TooltipIcon implements Events {
+    constructor(name: string, ariaText: string | null, elementShown: boolean, svgIcons?: Node[]) {
         Object.assign(this, Events);
```

Comment:
I know this is outside of type changes, but classes can extent Events. This might even simplify typing or remove the need for the interface:

 ```suggestion
class TooltipIcon extends Events {
    constructor(name: string, ariaText: string | null, elementShown: boolean, svgIcons?: Node[]) {
        super();
```